### PR TITLE
Names from nonexisiting objects

### PR DESCRIPTION
**Pull request by romanowski**
In some cases zinc reports dependencies to non-exiting objects that are taken from e.g. imports. This may later on manifest as dependencies to jar/classes from projects that should be handled as external dependencies.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?zinc=pull/422/head&deploy_to_bintray=false&run_sbt_community=false">this link.</A>

File: zinc/src/test/scala/sbt/inc/BinaryDepSpec.scala
Comment (eed3si9n) [inline zinc/src/test/scala/sbt/inc/BinaryDepSpec.scala]: Is this expectation true? What if you changed the macro? Shouldn't we invalidate the macro consumer?

Comment (romanowski) [reply zinc/src/test/scala/sbt/inc/BinaryDepSpec.scala]: We do not change macro here we compile each project once. 

We depend on NoopMacro but as a external dependency and we should not depended at the same time on it as library dependency (so we don't want recompile every usage of NoopMacro when we change anything in jar where it comes from)

**Comment by romanowski** [comment]
@eed3si9n @jvican @dwijnand Please let me know if you need any need anything more to review this PR :)

**Comment by jvican** [comment]
Myself I just need time. I'll get back to this once the big PRs in the queue are merged.

**Comment by dwijnand** [comment]
Where do we stand on this PR, @romanowski @eed3si9n @jvican?

It's green and merges cleanly. Can we land it?

**Comment by jvican** [comment]
I’m trying to review it this week.

**Review by jvican — APPROVED** [review]
LGTM

**Comment by jvican** [comment]
I'll merge when CI passes.

**Pull request merged**
